rumqttc/src/state.rs

```diff
@@ -309,6 +310,17 @@ impl MqttState {
         Ok(())
     }
 
+    fn outgoing_publish_with_notify(&mut self, publish: Publish, notify: OneshotSender<u16>) -> Result<(), StateError> {
+        let () = self.outgoing_publish(publish)?;
+        if let Some(Event::Outgoing(Outgoing::Publish(pkid))) = self.events.back() {
```

Comment:
This seems dangerous in in parallel environments. There is no guarantee that this will happen (and therefore the client could wait indefinetly for the notification) and even if this happens, there is a chance that the pkid is wrong.